app/src/main/java/com/worldtreeinc/leaves/form/ItemForm.java

```diff
@@ -122,22 +120,40 @@ public void getData(){
     }
 
     @Override
-    public void onClick(View v) {
+    public void onClick(final View v) {
         switch (v.getId()) {
             case R.id.item_image_select_icon:
                 new Thread(new Runnable() {
                     @Override
                     public void run() {
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M){
+                            if (activity.checkSelfPermission(Manifest.permission.CAMERA)== PackageManager.PERMISSION_GRANTED) {
+                                captureImage();
+                            } else {
+                                activity.requestPermissions(new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.CAMERA}, 150);
+                                Toast.makeText(activity,"Your Permission is needed to get access the camera",Toast.LENGTH_LONG).show();
+                            }
+                        } else {
                             captureImage();
-
+                        }
                     }
                 }).start();
                 break;
             case R.id.new_item_image:
                 new Thread(new Runnable() {
                     @Override
                     public void run() {
```

Comment:
NITPICK: this piece of conditional statement is the same as the one for camera. Try to extract to a separate function  which you can then call from the two separate places.